Watching a directory fails when it encounters a symbolic link to a nonexistent file

### Describe the bug

`fsnotify.Watcher.Add` with a directory instead of file (at least in `kqueue` implementation) ignores all files after the first symbolic link that points to a nonexistent file and returns a `file does not exist` error.

[This change](https://github.com/fsnotify/fsnotify/pull/681/changes#diff-cb5d0da6868fb1bd63f13e50565c5ca4951541c055c62707fc016f632e721f2eR413) is suspected.

### Code to Reproduce

### Code

```go
package main

import (
	"github.com/fsnotify/fsnotify"
)

func check(err error) {
	if err != nil {
		panic(err)
	}
}

func main() {
	w, err := fsnotify.NewWatcher()
	check(err)
	defer w.Close()

	err = w.Add("/another/directory")
	check(err)
}
```

### Expected result

Exit code: 0

### Actual result

```none
panic: "/another/directory/aaa": no such file or directory
```

### File operations to reproduce

> [!NOTE]
> These steps should be made before running the code.

1. 
    ```sh
    $ touch /some/directory/aaa
    ```
1.
    ```sh
    $ ln -s /another/directory/aaa /some/directory/aaa
    ```
1.
    ```sh
    $ rm /some/directory/aaa
    ```

### Which operating system and version are you using?

```none
ProductName:            macOS
ProductVersion:         26.1
BuildVersion:           25B78
```

### Which fsnotify version are you using?

1.9.0

### Did you try the latest main branch?

Yes

--- a/backend_kqueue.go
+++ b/backend_kqueue.go
@@ -578,12 +578,14 @@ func (w *kqueue) watchDirectoryFiles(dirPath string) error {
 
 		cleanPath, err := w.internalWatch(path, fi)
 		if err != nil {
-			// No permission to read the file; that's not a problem: just skip.
-			// But do add it to w.fileExists to prevent it from being picked up
-			// as a "new" file later (it still shows up in the directory
+			// No permission, or the entry resolved to a missing target
+			// (e.g. a dangling symlink): not a problem, just skip. But
+			// do mark it as seen to prevent it from being picked up as
+			// a "new" file later (it still shows up in the directory
 			// listing).
 			switch {
-			case errors.Is(err, unix.EACCES) || errors.Is(err, unix.EPERM):
+			case errors.Is(err, unix.EACCES) || errors.Is(err, unix.EPERM) ||
+				errors.Is(err, os.ErrNotExist):
 				cleanPath = filepath.Clean(path)
 			default:
 				return fmt.Errorf("%q: %w", path, err)

--- a/testdata/watch-dir/bug-727
+++ b/testdata/watch-dir/bug-727
@@ -14,6 +14,3 @@ touch /after
 
 Output:
 	create  /after
-
-	dragonfly:  # TODO: should fix this if possible.
-		no-events

--- a/testdata/watch-dir/bug-727
+++ b/testdata/watch-dir/bug-727
@@ -1,0 +1,19 @@
+# https://github.com/fsnotify/fsnotify/issues/727
+#
+# A dangling symlink already present in the directory must not cause
+# Watcher.Add to fail, and must not prevent sibling files from being
+# watched. Regression on kqueue since #681.
+require symlink
+
+ln   -s /nonexistent /dangling
+touch                /file
+
+watch /
+
+touch /after
+
+Output:
+	create  /after
+
+	dragonfly:  # TODO: should fix this if possible.
+		no-events

